File: java/app/src/main/java/net/fbvictorhugo/j/barreirasanitaria/data/dao/IPessoaDAO.java
package net.fbvictorhugo.j.barreirasanitaria.data.dao;

import net.fbvictorhugo.j.barreirasanitaria.data.model.Pessoa;

import java.util.List;

public interface IPessoaDAO {

    List<Pessoa> listar();

    void inserir(Pessoa pessoa) throws Exception;

    void atualizar(Pessoa pessoa) throws Exception;

}

File: kotlin/app/src/main/java/net/fbvictorhugo/k/barreirasanitaria/data/dao/IPessoaDAO.kt
package net.fbvictorhugo.k.barreirasanitaria.data.dao

import net.fbvictorhugo.k.barreirasanitaria.data.model.Pessoa

interface IPessoaDAO {

    fun listar(): ArrayList<Pessoa>

    @Throws(Exception::class)
    fun inserir(pessoa: Pessoa)

    @Throws(Exception::class)
    fun atualizar(pessoa: Pessoa)

    fun pesquisar(numeroDocumento: Long): ArrayList<Pessoa>

}
